specs/simple-serialize.md

```diff
@@ -101,9 +100,9 @@ We first define helper functions:
 
 We now define Merkleization `hash_tree_root(value)` of an object `value` recursively:
 
-* `merkleize(pack(value))` if `value` is a basic object or a vector of basic objects
+* `merkleize(pack(value))` if `value` is a basic object or a container of only basic objects
```

Comment:
I'm not sure packing of non-homogeneous basic-type containers is particularly clean. Maybe we do want to restrict packing to vectors.

Reply:
yeah, I think I agree. Reverted to the original logic, but changed the phrasing to not use `vector` anymore (to avoid confusion if an object is not explicitly defined to be a vector, but has the same structure).

Reply:
I'm not sure it's a win anymore. The reason is that if developers want packing of their homogenous containers (something we don't have anywhere in the Eth2.0 spec!), then just pick a vector. And for most languages (e.g. Golang) there's a natural distinction between containers (`struct`) and vectors (`slice`) which breaks down because now you have to do this special check on `struct`s to see if they are actually a `slice`.

Reply:
@jannikluhn Do you still want to pursue this PR?

Reply:
@jannikluhn Do you still want to pursue this PR?

Reply:
no, I don't think this is particularly important.